core/state_transition.go

```diff
@@ -244,7 +246,7 @@ func (st *StateTransition) buyGas() error {
 	mgval := new(big.Int).SetUint64(st.msg.GasLimit)
 	mgval = mgval.Mul(mgval, st.msg.GasPrice)
 	var l1Cost *big.Int
-	if st.evm.Context.L1CostFunc != nil && !st.msg.SkipAccountChecks {
+	if st.evm.Context.L1CostFunc != nil && !st.msg.SkipAccountChecks && !st.msg.IsMintTokenTx {
 		l1Cost = st.evm.Context.L1CostFunc(st.evm.Context.BlockNumber.Uint64(), st.msg.RollupDataGas, st.msg.IsDepositTx)
```

Comment:
Isn't it better? Then `NewL1CostFunc` should be changed also.
```suggestion
	if st.evm.Context.L1CostFunc != nil && !st.msg.SkipAccountChecks {
		l1Cost = st.evm.Context.L1CostFunc(st.evm.Context.BlockNumber.Uint64(), st.msg.RollupDataGas, st.msg.IsDepositTx || st.msg.IsMintTokenTx)
```